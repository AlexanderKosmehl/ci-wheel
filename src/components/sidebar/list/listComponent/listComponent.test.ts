import {
  describe, expect, it, vi,
} from 'vitest';
import generateListComponent from './listComponent';

describe('generateListComponent', () => {
  const entryRemovalCallback = vi.fn((_removedEntry: string) => {});
  const listEntries = ['Test', 'Test2'];

  const listComponent = generateListComponent({
    listEntries,
    entryRemovalCallback,
  });

  it('generates component correctly', () => {
    expect(listComponent).toMatchSnapshot();
  });

  it('adds working removal callback', () => {
    const listElement = listComponent.querySelector<HTMLLIElement>('li');

    if (!listElement) throw Error('No list entries rendered!');

    listElement.querySelector<HTMLButtonElement>('[data-test=listEntryDeleteButton]')?.click();
    expect(entryRemovalCallback).toBeCalled();
    expect(entryRemovalCallback).toBeCalledWith('Test');
  });
});
