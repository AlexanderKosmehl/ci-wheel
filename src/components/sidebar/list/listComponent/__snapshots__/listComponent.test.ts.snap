// Vitest Snapshot v1

exports[`generateListComponent > generates component correctly 1`] = `
<ul
  class="listContainer"
>
  <li
    class="container"
  >
    <span
      class="label"
    >
      Test
    </span>
    <button
      class="button deleteButton"
      data-test="listEntryDeleteButton"
    >
      <img
        class="icon"
        src="/src/assets/delete-icon.svg"
      />
    </button>
  </li>
  <li
    class="container"
  >
    <span
      class="label"
    >
      Test2
    </span>
    <button
      class="button deleteButton"
      data-test="listEntryDeleteButton"
    >
      <img
        class="icon"
        src="/src/assets/delete-icon.svg"
      />
    </button>
  </li>
</ul>
`;
